### ISSUE TITLE
Results don't make sense after applying augmentations 

### ISSUE THREAD

**Issue by mansi-aggarwal-2504** [matterport/Mask_RCNN#2663]
I am using Mask RCNN on custom dataset. I trained using `resnet101` backbone, with 150 training images and 50 validation images. 70 epochs and 50 steps per epoch. But I also saved the best model which was at epoch 40. The results on val dataset were average. But at least the masks created were passable.

I applied augmentations to improve results. All other parameters were kept untouched.
```
augmentation = iaa.SomeOf((0, None), [
iaa.Fliplr(1),
iaa.Flipud(1),
iaa.OneOf([iaa.Affine(rotate = 90 * i) for i in range(1, 3)]),
iaa.OneOf([iaa.Affine(rotate = -90 * i) for i in range(1, 3) ]),
iaa.Affine(scale={"x": (0.5, 1.5), "y": (0.5, 1.5)}),
iaa.sizelib.CropAndPad(percent=([0.05, 0.1], [0.05, 0.1], [0.05, 0.1], [0.05, 0.1])),
])
```

But now, there are masks on hardly 10% of the validation images which don't make sense at all. The masks include big rectangle patch covering vague parts of images (30-40% of the image sometimes), whereas my image includes multiple circular objects. I am not sure what is going wrong, the training loss was 1.7 and dropped to 1.3-1.2 over epochs but the validation loss hovered over ~2 but didn't improve much over time. Can someone help with what I can do to improve results?

**Comment by OBrink** [comment]
I checked the augmentations on some images in the balloon dataset and can confirm that the applied augmentations don't break anything. Did you change anything else?

**Comment by mansi-aggarwal-2504** [comment]
Hello @OBrink, first of all, thank you for this [notebook](https://github.com/OBrink/Mask_RCNN/blob/master/samples/balloon/inspect_balloon_data.ipynb). I used this to inspect my augmentations which are working fine.
Apart from that, no, I didn't change anything at all. Just added the above mentioned augmentations.

**Comment by mansi-aggarwal-2504** [comment]
@OBrink correction. I trained without augmentation using `resent101` and with augmentation using `resent50`. Since I couldn't run a deeper network on my machine. I got a machine with higher GPU config and used `resnet101` with augmentations since I thought only the `BACKBONE` had been different but this time, the validation loss shot up and hovered over ~8.
Yet to check the predictions however unlikely that I'd get good masks. 
Can someone tell me what the problem could be? I want to improve my masks for which I applied augmentations but that's not helping, resultant masks don't make sense at all.
